daemon/container_operations.go

```diff
@@ -379,21 +381,26 @@ func (daemon *Daemon) findAndAttachNetwork(container *container.Container, idOrN
 		retryCount int
 	)
 
+	if n == nil && daemon.attachableNetworkLock != nil {
+		daemon.attachableNetworkLock.Lock(id)
+		defer daemon.attachableNetworkLock.Unlock(id)
+	}
+
```

Comment:
Honestly, the `n != nil` is an optimization since if it's not nil, then there is no need to lock.
I also despise `else` blocks. 🤷‍♂️ 
The reason to not implicitly lock is because this is extra serialization when it's not really needed.

Reply:
*nod*  I get that the lock isn't needed if n != nil.   Was just pointing out that that test was already effectively done above with the err != nil, so the body could be moved up there (w/ test for lock existence).  Keeps the "network not found" logic in one place.

The if / else was a side-comment re: the way the existing code is already structured.   I get the preference for avoiding "else" clauses in general.  As a fresh reader of the code, I was expressing that it might be better in this case because `if (x) {...} if (y) {...}` leads me to believe that both `x` and `y` could be true at the same time for purposes of subsequent logic.  (and that is *not* the case here)  In any case it's immaterial for purposes of this fix.

Reply:
Generally, if an early-return is possible, I'd try to avoid an `else`. I do agree that these "ifs" are a bit "iffy" 😅, e.g.: https://github.com/moby/moby/blob/c379d2681ffe8495a888fb1d0f14973fbdbdc969/daemon/container_operations.go#L352-L358

could be simplified to:

```go
if err != nil && container.Managed {
	// We should always be able to find the network for a managed container.
	return nil, nil, err
}
```

The second `if` would become;

```go
// If we found a network and if it is not dynamically created
// we should never attempt to attach to that network here.
if n != nil && (container.Managed || !n.Info().Dynamic()) {
	return n, nil, nil
}
```

Which leads to an interesting question: if the container is _managed_, we never attach, so, should this simply be this?:


```go
if container.Managed {
	return n, nil, err
}
```

Reply:
The above simplification is very appealing.   Presumably it would be followed by an:
```go
if n != nil && !n.Info().Dynamic() {
        return n, nil, nil
}
```
to handle the second test?  Again, would be much more readable (to me anyway).

Reply:
@ctelfer-docker I agree (but also felt it being separate from this PR): would you be interested in opening a PR for that?

Reply:
Agree and will do.